File: Medium/40BestSet.py
def bestSeat(seats):
    # Write your code here.
    bestSeat = -1
    maxSpace = 0
    left = 0
    while left < len(seats):
        right = left+1
        while right < len(seats) and seats[right] == 0:
            right += 1

        availableSpace = right-left-1
        if availableSpace > maxSpace:
            bestSeat = (right+left) // 2
            maxSpace = availableSpace
        left = right
    return bestSeat


print(bestSeat([1, 0, 1, 0, 0, 0, 1]))

File: Medium/40BestSet.js
function bestSeat(seats) {
  // Write your code here.
  let bestSeat = -1;
  let maxSpace = 0;

  let left = 0;
  while (left < seats.length) {
    let right = left + 1;
    while (right < seats.length && seats[right] === 0) {
      right++;
    }
    let currentLength = right - left - 1;
    if (currentLength > maxSpace) {
      bestSeat = Math.floor((right + left) / 2);
      maxSpace = currentLength;
    }
    left = right;
  }
  return bestSeat;
}
console.log(bestSeat([1, 0, 1, 0, 0, 0, 1]));
